simba/report.py

```diff
@@ -360,3 +381,259 @@ def write_csv(data: Iterable, file_path, propagate_errors=False):
+    plt.close()
+
+
+def plot_consumption_per_rotation_distribution(extended_plots_path, schedule):
+    """Plots the distribution of bus types in consumption brackets as a stacked bar chart.
+
+    :param extended_plots_path: directory to save plot to
+    :type extended_plots_path: Path
+    :param schedule: Driving schedule for the simulation, schedule.rotations are used
+    :type schedule: eBus-Toolbox.Schedule
+    """
+
+    step = 50
+    # get number of bins
+    max_con = int(max([schedule.rotations[rot].consumption for rot in schedule.rotations]))
+    num_bins = max_con // step + 1  # number of bars in plot
+    labels = [f"{i*step} - {(i+1)*step}" for i in range(num_bins)]
+    # init bins: track consumption bins for each vehicle type individually
+    bins = {v_types: [0]*num_bins for v_types in schedule.vehicle_types}
+
+    # fill bins with rotations
+    for rot in schedule.rotations:
+        position = int(schedule.rotations[rot].consumption // step)
+        bins[schedule.rotations[rot].vehicle_type][position] += 1
+
+    # plot
+    fig, ax = plt.subplots()
+    bar_bottom = [0] * num_bins
+    for v_type in schedule.vehicle_types:
+        ax.bar(labels, bins[v_type], width=0.9, label=v_type, bottom=bar_bottom)
+        for i in range(num_bins):
+            bar_bottom[i] += bins[v_type][i]
+    ax.set_xlabel('Energy consumption [kWh]')
+    plt.xticks(rotation=30)
+    ax.set_ylabel('Number of rotations')
+    ax.yaxis.get_major_locator().set_params(integer=True)
+    ax.yaxis.grid(True)
+    ax.set_title('Distribution of bus types over rotation energy consumption')
+    ax.legend()
+    plt.savefig(extended_plots_path / "distribution_bustypes_consumption_rotations")
+    plt.close()
+
+
+def plot_charge_type_distribution(extended_plots_path, scenario, schedule):
+    """Plots the number of rotations of each charging type in a bar chart.
+
+    :param extended_plots_path: directory to save plot to
+    :type extended_plots_path: Path
+    :param scenario: Scenario for with to generate timeseries.
+    :type scenario: spice_ev.Scenario
+    :param schedule: Driving schedule for the simulation. schedule.rotations are used
+    :type schedule: eBus-Toolbox.Schedule
+    """
+    # count charging types (also with regard to negative rotations)
+    charging_types = {'oppb': 0, 'oppb_neg': 0, 'depb': 0, 'depb_neg': 0}
+    negative_rotations = schedule.get_negative_rotations(scenario)
+    for rot in schedule.rotations:
+        ct = schedule.rotations[rot].charging_type
+        if rot in negative_rotations:
+            ct += '_neg'
+        try:
+            charging_types[ct] += 1
+        except KeyError:
+            logging.error(f"Rotation {rot}: unknown charging type: '{ct}'")
+
+    # plot
+    fig, ax = plt.subplots()
+    bars1 = ax.bar(
+        ["Opportunity", "Depot"],
+        [charging_types["oppb"], charging_types["depb"]],
+        color=["#66CDAA", "#66CDAA"],
```

Comment:
why these colors? please use verbose names like light_green, if the colors are picked for this reason. I'd recommend staying in the default matplotlib color scheme or switching the matplotlib colorscheme to a rli specific colorscheme,

Reply:
Took these from original code. Now reverted to default colors.